bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/action/ILGWebOSActions.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgwebos.action;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link ILGWebOSActions} defines the interface for all thing actions supported by the binding.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public interface ILGWebOSActions {
+
+    public void showToast(String text) throws IOException;
+
+    public void showToast(String icon, String text) throws IOException;
+
+    public void launchBrowser(String url);
+
+    public List<Application> getApplications();
```

Comment:
I don't think the bugfix would work here. The proxy workaround would only work for methods that have parameters and return types that are loaded in a 3rd untouched classloader (e.g. parent or core). The `Application` class is loaded as part of the bundle classloader the bug would still be present when `LGWebOSActions` attempts to use Application instances returned from `getApplications()`.

One thing to note is that `getApplications` isn't annotated with `@RuleAction` which would mean the `getApplications` static method doesn't have a corresponding RuleAction. I'm not sure why this would be the case.
@sprehn can you give an explanation?

Reply:
Ok, even if we could still have a problem with `getApplications`, at least it will be ok for all other actions.
As we are on a temporary workaround, I propose to accept this limit and mention it with a comment in the code.
Is it ok for you ?

For the missing `@RuleAction `, good catch, I hope @sprehn is still available to answer this question.

Reply:
Comment added for getApplications

Reply:
Hi, this method's intend is to allow users to find out what the Ids of applications are. It is not designed to work with the new rule engine. I think it is now safe to remove it all together from the action, as the binding can now be queried for the same information via karaf console.

Reply:
Done, action removed.

Reply:
super, pls do me the favor and also delete the section in the readme file.
https://www.openhab.org/addons/bindings/lgwebos/#list-getapplications

Reply:
since this is merged already, see #8161